tests/invalid_models_tests/test_models.py

```diff
@@ -761,3 +761,75 @@ class ParkingLot(Place):
             'as an implicit link is deprecated.'
         )
         self.assertEqual(ParkingLot._meta.pk.name, 'parent')
+
+    def test_m2m_table_name_clash(self):
+        class Foo(models.Model):
+            bar = models.ManyToManyField('Bar', db_table='myapp_bar')
+
+            class Meta:
+                db_table = 'myapp_foo'
+
+        class Bar(models.Model):
+            class Meta:
+                db_table = 'myapp_bar'
+
+        errors = Foo.check()
+        expected = [
+            Error(
+                "Field is using a table that has already been registered by "
+                "'invalid_models_tests.Bar'.",
+                obj=Foo._meta.get_field('bar'),
+                id='fields.E340',
+            )
+        ]
+        self.assertEqual(errors, expected)
+
+    def test_m2m_field_table_name_clash(self):
+        class Foo(models.Model):
+            pass
+
+        class Bar(models.Model):
+            foos = models.ManyToManyField(Foo, db_table='clash')
+
+        class Baz(models.Model):
+            foos = models.ManyToManyField(Foo, db_table='clash')
+
+        errors = Bar.check() + Baz.check()
+        expected = [
+            Error(
+                "Field is using a table that has already been registered by "
+                "'invalid_models_tests.Baz.foos'.",
+                obj=Bar._meta.get_field('foos'),
+                id='fields.E340',
+            ),
+            Error(
+                "Field is using a table that has already been registered by "
+                "'invalid_models_tests.Bar.foos'.",
+                obj=Baz._meta.get_field('foos'),
+                id='fields.E340',
+            )
+        ]
+        self.assertEqual(errors, expected)
+
+    def test_m2m_autogenerated_table_name_clash(self):
+        class Foo(models.Model):
+            class Meta:
+                db_table = 'bar_foos'
+
+        class Bar(models.Model):
+            # In this case the auto-generated `db_table` will be bar_foos
+            foos = models.ManyToManyField(Foo)
+
+            class Meta:
+                db_table = 'bar'
+
+        errors = Bar.check()
+        expected = [
+            Error(
+                "Field is using a table that has already been registered by "
+                "'invalid_models_tests.Foo'.",
+                obj=Bar._meta.get_field('foos'),
+                id='fields.E340',
+            )
+        ]
+        self.assertEqual(errors, expected)
```

Comment:
Do you still want me to remove `errors = ...` from test_m2m_table_name_clash and test_m2m_autogenerated_table_name_clash or are they OK to be compatible with the rest of the file?